### DISCUSSION TITLE
Vim navigation on filtered view

### DISCUSSION THREAD

**Discussion by IvanVolosyuk** [nvim-neo-tree/neo-tree.nvim#584] [General]
I'm trying to migrate to nvim/neo-tree from vim/nerdtree. The concept of filtered view is very cool and I'm trying to get used to it, but I guess I don't understand some principles.

In one specific example I'm trying to do the filtering and display just 'init.lua' files:
```/init.lua```
For some other name it would be much better, but not for this particular case, see the picture below. Now I would like to accept the filtered view and switch to vim navigation mode - jump at the particular line or at the bottom of the list in this case: ``G``, but I cannot. The intuitive choice after ``/`` vim search is ``<CR>`` in order to go from incremental search to completed search. But ``<CR>`` just opens the file at the cursor. Do I miss something obvious or filtered view doesn't work with vim navigation keys in neo-tree? I discovered that I can use ``<C-n>``, ``<C-p>`` but it is not helpful in this case as it doesn't make me faster in opening the last file in the list.
**[Update]:** I found out I can use ``f`` to filter on submit, but can I press a key on increment search to make it stick?
**[Update2]:** It looks like I can change the config to make ``<CR>`` act as search completion and allow navigation in filtered view:
```
filesystem = {
    window = {
      mappings = {
         ["/"] = "filter_as_you_type",
     }
   }
}
```
Well it changes the question :).  Can I use the best of the both worlds, say use ``fuzzy_finder`` with:

-   ``<CR>`` - opens the file under cursor
-   ``<C-y>`` - submits the filtering view similar as ``<CR>`` for ``filter_as_you_type`` and allows to use vim keys for navigation

![init](https://user-images.githubusercontent.com/7278365/198815059-8603dd94-8ece-4d2e-86aa-3616f2e2da63.png)

**Comment by teddywaweru** [comment]
My suggestion would be that the `/` function, upon pressing `<CR>`, should revert to browsing through the filter results similarly to `f` upon submit. It's a rare circumstance that the file being filtered for will be the first one in the browser area. Also, if it is, it would only be an extra `<CR>` click away... 

PS: I'm currently using AstroNvim.
